src/components/FileItem/FileItem.tsx

```diff
@@ -1,22 +1,83 @@
-import React from "react";
-import { useParams } from "react-router-dom";
-import { useGetFileByNameQuery } from "../../services/task1";
-import { Box, Card, Typography, useTheme } from "@mui/material";
+import React, { useEffect, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import {
+  useChangeFileMutation,
+  useCreateFileMutation,
+  useDeleteFileMutation,
+  useGetFileByNameQuery,
+} from "../../services/task1";
+import { Box, Button, Card, Grid, TextField, Typography } from "@mui/material";
 
 export const FileItem = () => {
   const { filename } = useParams();
+  const navigate = useNavigate();
+
   const { data, error, isLoading } = useGetFileByNameQuery(filename!);
-  const theme = useTheme();
+  const [changeFile, {}] = useChangeFileMutation();
+  const [deleteFile, {}] = useDeleteFileMutation();
+
+  const [content, setContent] = useState("");
+  const [editMode, setEditMode] = useState(false);
+
+  useEffect(() => {
+    setContent((data || {}).content || "");
```

Comment:
you can use optional chaining here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining instead of `|| {}`